### PR TITLE
samples: nrf9160: lwm2m_client: Add bootstrap support to the sample

### PR DESCRIPTION
**Pull request by rlubos**
Add bootstrap support to the `lwm2m_client` sample. Provision PSK/PSK ID received during the bootsrap procedure into the modem.

`DNM` until Zephyr dependencies are merged and introduced into `sdk-zephyr`.

### CODE REVIEW

**Comment by rlubos** [comment]
FYI @joakimtoe

**Review by umapraseeda — APPROVED** [review]

File: samples/nrf9160/lwm2m_client/src/main.c [hunk 1]
```diff
@@ -142,6 +150,161 @@ static int lwm2m_setup(void)
...
+	lwm2m_rd_client_start(&client, endpoint_name, false, rd_client_event);
+}
+#endif
+
+static void provision_credentials(void)
+{
+	int security_instance;
+	int security_mode;
+
+
```
Comment (lemrey) [inline samples/nrf9160/lwm2m_client/src/main.c:277]: Extra newline

**Review by lemrey — APPROVED** [review]

**Comment by tejlmand** [comment]
Please update SHA: d14bfb6a9e06541842ecb73e042b076ff98ed59f

**Comment by rlubos** [comment]
SHA updated

**Review by tejlmand — APPROVED** [review]
lgtm

**Comment by JusbeR** [comment]
Do you @rlubos have any more information about the [CC301 crash workaround](https://github.com/nrfconnect/sdk-nrf/pull/2997/commits/bfdbc198884f9b5ba4f10c2d53c5e529cbfbdbe9) that is merged in this PR?

I am asking because it indeed crashes for my application too after upgrading to 1.4 release and disabling it it not the best thing that one can do in production app.

**Comment by rlubos** [comment]
@JusbeR unfortunately SPM entropy issue is still being investigated internally. There were some fixes in 1.4. but apparently, they weren't enough. It is suspected it might be a stack overflow in SPM, increasing the stack prevents the crash. The SPM stack is already increased on the master.

**Pull request merged**